sdk/eventgrid/azure-eventgrid/azure/eventgrid/_models.py

```diff
@@ -106,6 +107,31 @@ def __init__(self, source, type, **kwargs): # pylint: disable=redefined-builtin
             raise ValueError("data and data_base64 cannot be provided at the same time.\
                 Use data_base64 only if you are sending bytes, and use data otherwise.")
 
+    @classmethod
+    def from_dict(cls, event, **kwargs):
+        # type: (Dict, Any) -> CloudEvent
+        """
+        Returns the deserialized CloudEvent object when a dict is provided.
+
+        :param event: The dict representation of the event which needs to be deserialized.
+        :type event: dict
```

Comment:
I think it would be helpful if we could add description on the key-value items we accepts -- but it's a doc stuff

Reply:
hmm, the payload normally comes directly from a storage queue - i can put a link to the relevant spec

Reply:
If not all the key-value pairs, at least mentioning that `data_base64` will always be decoded and set as the value for the `data` field would be helpful. In general, maybe mentioning that `from_dict` will never result in the `data_base64` field being set in the resulting `CloudEvent` object (if I'm getting that right).